plugins/outputs/socket_writer/socket_writer_test.go

```diff
@@ -44,9 +44,10 @@ func TestSocketWriter_udp(t *testing.T) {
 }
 
 func TestSocketWriter_unix(t *testing.T) {
-	defer os.Remove("/tmp/telegraf_test.sock")
+	os.Remove("/tmp/telegraf_test.sock")
 	listener, err := net.Listen("unix", "/tmp/telegraf_test.sock")
 	require.NoError(t, err)
+	defer os.Remove("/tmp/telegraf_test.sock")
```

Comment:
I think the defer needs to move upwards to ensure it is ran even if the require.NoError fails.